File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/util/SoftKeyboard.java
package de.dimedis.mobileentry.util;

import android.app.Service;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

public class SoftKeyboard {

    public static void showKeyboard(View view) {
        setShowKeyboard(view, true);
    }

    public static void hideKeyboard(View view) {
        setShowKeyboard(view, false);
    }

    public static void setShowKeyboard(View view, boolean isShow) {
        if (view == null) return;
        InputMethodManager imm = (InputMethodManager) view.getContext().getSystemService(Service.INPUT_METHOD_SERVICE);
        if (isShow) imm.showSoftInput(view, InputMethodManager.SHOW_IMPLICIT);
        else imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/util/SoftKeyboard.kt
package de.dimedis.mobileentry.util

import android.app.Service
import android.view.View
import android.view.inputmethod.InputMethodManager

object SoftKeyboard {
    fun showKeyboard(view: View?) {
        setShowKeyboard(view, true)
    }

    fun hideKeyboard(view: View?) {
        setShowKeyboard(view, false)
    }

    fun setShowKeyboard(view: View?, isShow: Boolean) {
        if (view == null) return
        val imm = view.context.getSystemService(Service.INPUT_METHOD_SERVICE) as InputMethodManager
        if (isShow) imm.showSoftInput(view, InputMethodManager.SHOW_IMPLICIT)
        else imm.hideSoftInputFromWindow(view.windowToken, 0)
    }
}